execution.bs

```diff
@@ -3173,67 +3196,91 @@ Insert this section as a new subclause between Header `<stop_token>` synopsis <b
+3. Let `t` and `u` be distinct, valid objects of type `T` and let `init` be an
+    object of type `Initializer`. Then for some type `CB`, the type `T` models
+    `stoppable_token_for<CB, Initializer>` only if:
 
-    4. Given a callback-type, CB, and a callback-initializer argument, `init`, of type `Initializer` then constructing an instance, `cb`, of type `T::callback_type<CB>`, passing `t` as the first argument and `init` as the second argument to the constructor, shall,
-        if `t.stop_possible()` is `true`, construct an instance, `callback`, of type `CB`, direct-initialized with `init`, and register callback with `t`’s shared stop state such that callback will be invoked with an empty argument list if a stop request is made on
-        the shared stop state.
+    1. Constructing an instance, `cb`, of type `T::callback_type<CB>`, passing
+        `t` as the first argument and `init` as the second argument to the
+        constructor, shall, if `t.stop_possible()` is `true`, construct an
+        instance, `callback`, of type `CB`, direct-initialized with `init`,
+        and register callback with `t`'s shared stop state such that `callback`
+        will be invoked with an empty argument list if a stop request is made
+        on the shared stop state.
 
-        1. If `t.stop_requested()` is `true` at the time callback is registered then callback may be invoked immediately inline inside the call to `cb`’s constructor.
+        1. If `t.stop_requested()` is `true` at the time `callback` is
+            registered then `callback` may be invoked immediately inline inside
+            the call to `cb`'s constructor.
 
-        2. If callback is invoked then, if `u` references the same shared stop state as `t`, an evaluation of `u.stop_requested()` will be `true` if the beginning of the invocation of callback strongly-happens-before the evaluation of `u.stop_requested()`.
+        2. If `callback` is invoked then, if `u` references the same shared stop
+            state as `t`, an evaluation of `u.stop_requested()` will be `true`
+            if the beginning of the invocation of `callback`
+            strongly-happens-before the evaluation of `u.stop_requested()`.
 
-        3. If `t.stop_possible()` evaluates to `false` then the construction of `cb` is not required to construct and initialize `callback`.
+        3. [*Note:* If `t.stop_possible()` evaluates to `false` then the construction of
+            `cb` is not required to construct and initialize `callback`. *--end note*]
 
-    5. Construction of a `T::callback_type<CB>` instance shall only throw exceptions thrown by the initialization of the `CB` instance from the value of type `Initializer`.
+    2. Construction of a `T::callback_type<CB>` instance shall only throw exceptions thrown by the initialization of the `CB` instance from the value of type `Initializer`.
```

Comment:
Should this require that  `std::is_nothrow_constructible_v<typename T::template callback_type<CB>, T, Initializer>` is `true` if `std::is_nothrow_constructible_v<CB, Initializer>` is `true`?

Reply:
No strong feelings on this one.

Reply:
`stop_callback`  allows callable, that has constructor marked as not throwing, or throwing. I do not think we want the concept to exclude it.